### DISCUSSION TITLE
Alpha Program and Self Signed Certificate experience on iPad (OS 14.1, Chrome 85.0.4183.109)

### DISCUSSION THREAD

**Discussion by mboyar** [coder/code-server#2350] [Show and tell]
Hello everyone,

I’ve applied [Self Signed Certificate](https://github.com/cdr/code-server/blob/v3.7.2/doc/guide.md#self-signed-certificate) steps _**without**_ using [ipad.md](https://github.com/cdr/code-server/blob/v3.7.2/doc/ipad.md), and I am running code-server with “—link” argument in systemd file, and accessing over HTTPS  to ***.cdr.co. Thank you very much for this nice work.

If I am missing something, please tell me.

Murat

**Comment by code-asher** [comment]
Awesome! Very glad to see it's working well. 

`--link` will actually disable `--cert` and listen on a random port so you should be able to use it without applying any of the self-signed certificate steps.

**Reply by mboyar** [reply]
yes, but if no cert at that time I've seen the notice pop-up at the right-bottom-corner about the restrictions on copy-paste and similar capabilities.

**Reply by code-asher** [reply]
If you use `--link` and visit through *.cdr.co then it will use our certificate and not the local one. Actually code-server should say `Not serving HTTPS` which means it's not using local certificates. If it does use local certificates then it'll say `Using certificate for HTTPS`.

Lemme know if I've misunderstood! It's very possible I've missed something. :smile:

**Reply by code-asher** [reply]
I just realized the URL we output uses `http` though. `--link` will serve both `http` and `https` without needing any local certificates but we should change that output to use `https` or automatically redirect from `http` to `https`. Going to the `http` version is probably where you saw the notice about the clipboard, etc.
